### ISSUE TITLE
Access Denied Exception when creating simple planned task

### ISSUE THREAD

**Issue by dahall** [dahall/TaskScheduler#192]
Hello,

I'm creating a simple planned task like this:

```
            using (TaskService ts = new TaskService())
            {
                TaskDefinition td = ts.NewTask();
                td.RegistrationInfo.Description = "AutoRestartTest";

                LogonTrigger mainTrigger = new LogonTrigger();
                mainTrigger.ExecutionTimeLimit = TimeSpan.FromDays(30);
                mainTrigger.Repetition.Interval = TimeSpan.FromMinutes(1);

                td.Triggers.Add(mainTrigger);

                td.Actions.Add(new ExecAction(@"C:\example.exe", null, null));

                ts.RootFolder.RegisterTaskDefinition(@"TestTask1", td);
             }
```

And I am getting this exception at the last line of the code:

Access Denied (Exception of HRESULT: 0x80070005 (E_ACCESSDENIED))

I have created a Manifest-File to take care of this option being set to this:

```
        <requestedExecutionLevel level="highestAvailable" uiAccess="false" />
```

Please help!
What am I doing wrong?

Greetings,

Tom

ps. I dont know why code inserted in the codeblock remains black/white.

Originally posted: 2016-04-11T11:09:13

**Comment by dahall** [comment]
Please look over the information at https://taskscheduler.codeplex.com/wikipage?title=TaskSecurity. My guess is that the LogonTrigger with no user specified requires higher permissions than the running process has. Please let me know if that page answers your question. Also check the Microsoft pages for more information on registering tasks. The error you're seeing is from the native Microsoft library and not this wrapper.

Originally posted: 2016-04-11T11:51:27

**Comment by dahall** [comment]
Hello,

Thanks for your quick reply.

Sadly, the link doesnt help. I want to create a planned task for the user who is running my application. The site from your link only explains different situations.

Is there an example which does create a task for the current user who runs the application?

This should be the easiest use of the wrapper.

Greetings,

Tom



Originally posted: 2016-04-12T06:52:22

**Comment by dahall** [comment]
Okay I found out that it had to do with the fact that I'm defining a LogonTrigger without setting its "UserId" property. When I set it to the currently logged on user, it entered the task to the task scheduler and didnt throw the exception.

I guess it has something to do with when the UserId isnt defined, in the Task-scheduler you would choose the option "at the logon of every user" which surely depends on user rights.

Originally posted: 2016-04-12T11:01:07
